core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java

```diff
@@ -4262,6 +4262,87 @@ public void testLatestByAllValueIndexedColumn() throws Exception {
         });
     }
 
+    @Test
+    public void testLatestByFilteredBySymbolInNoIndexes() throws Exception {
+        testLatestByFilteredBySymbolIn("create table x (\n" +
+                "  ts timestamp,\n" +
+                "  node symbol,\n" +
+                "  metric symbol,\n" +
+                "  value long) \n" +
+                "  timestamp(ts) partition by day");
+    }
+
+    @Test
+    public void testLatestByFilteredBySymbolInAllIndexed() throws Exception {
+        testLatestByFilteredBySymbolIn("create table x (\n" +
+                "  ts timestamp,\n" +
+                "  node symbol index,\n" +
+                "  metric symbol index,\n" +
+                "  value long) \n" +
+                "  timestamp(ts) partition by day");
+    }
+
+    @Test
+    public void testLatestByPartiallyIndexedFilteredSymbolIn() throws Exception {
+        testLatestByFilteredBySymbolIn("create table x (\n" +
+                "  ts timestamp,\n" +
+                "  node symbol index,\n" +
+                "  metric symbol,\n" +
+                "  value long) \n" +
+                "  timestamp(ts) partition by day");
+    }
+
+    private void testLatestByFilteredBySymbolIn(String ddl) throws Exception {
+        assertMemoryLeak(() -> {
+            compiler.compile(ddl, sqlExecutionContext);
+
+            executeInsert("insert into x values ('2021-11-17T17:35:01.000000Z', 'node1', 'cpu', 100)");
+            executeInsert("insert into x values ('2021-11-17T17:35:01.000000Z', 'node2', 'cpu', 75)");
```

Comment:
Would be good to have here several lines with `node2, cpu` to check that latest is taken.

Reply:
Good idea. Done in 265fd52e4de65e3d66e076a791ea44c0452b272b

Reply:
This one should contain the link: 265fd52e4de65e3d66e076a791ea44c0452b272b